ext/strings_test.go

```diff
@@ -329,3 +331,398 @@ func TestStringsWithExtension(t *testing.T) {
+			format:     "double is %x",
+			formatArgs: "0.5",
+			err:        "error during formatting: only integers, byte buffers, and strings can be formatted as hex, was given double",
+		},
+	}
+	runCase := func(format, formatArgs, locale string, t *testing.T) (ref.Val, error) {
+		var env *cel.Env
+		var err error
+		if locale != "" {
+			env, err = cel.NewEnv(cel.Locale(locale), Strings())
+		} else {
+			env, err = cel.NewEnv(Strings())
+		}
+		if err != nil {
+			t.Fatalf("cel.NewEnv(Strings()) failed: %v", err)
+		}
+		expr := fmt.Sprintf("%q.format([%s])", format, formatArgs)
+		parsedAst, issues := env.Parse(expr)
+		if issues.Err() != nil {
+			t.Fatalf("env.Parse(%v) failed: %v", expr, issues.Err())
+		}
+		checkedAst, issues := env.Check(parsedAst)
+		if issues.Err() != nil {
+			t.Fatalf("env.Check(%v) failed: %v", expr, issues.Err())
+		}
+		program, err := env.Program(checkedAst)
+		if err != nil {
+			t.Fatal(err)
+		}
+		out, _, err := program.Eval(cel.NoVars())
+		return out, err
+	}
+	checkCase := func(output ref.Val, expectedOutput string, err error, expectedErr string, t *testing.T) {
+		if err != nil {
+			if expectedErr != "" {
+				if err.Error() != expectedErr {
+					t.Fatalf("expected %q as error message, got %q", expectedErr, err.Error())
+				}
+			} else {
+				t.Fatalf("unexpected error: %s", err)
+			}
+		} else {
+			if output.Type() != types.StringType {
+				t.Fatalf("expected test expr to eval to string (got %s instead)", output.Type().TypeName())
+			} else {
+				outputStr := output.Value().(string)
+				if outputStr != expectedOutput {
+					t.Errorf("expected %q as output, got %q", expectedOutput, outputStr)
+				}
+			}
+		}
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			out, err := runCase(tt.format, tt.formatArgs, tt.locale, t)
+			checkCase(out, tt.expectedOutput, err, tt.err, t)
+			if tt.locale == "" {
+				// if the test has no locale specified, then that means it
+				// should have the same output regardless of lcoale
+				t.Run("no change on locale", func(t *testing.T) {
+					out, err := runCase(tt.format, tt.formatArgs, "da_DK", t)
+					checkCase(out, tt.expectedOutput, err, tt.err, t)
+				})
+			}
+		})
+	}
+}
+
+func TestBadLocale(t *testing.T) {
+	_, err := cel.NewEnv(cel.Locale("bad-locale"), Strings())
+	if err != nil {
+		if err.Error() != "failed to parse locale: language: subtag \"locale\" is well-formed but unknown" {
+			t.Errorf("expected error messaged to be \"failed to parse locale: language: subtag \"locale\" is well-formed but unknown\", got %q", err)
+		}
+	} else {
+		t.Error("expected NewEnv to fail during locale parsing")
+	}
+}
+
+func TestLiteralOutput(t *testing.T) {
+	tests := []struct {
+		name          string
+		formatLiteral string
+		expectedType  string
+	}{
+		{
+			name:          "map literal support",
+			formatLiteral: `{"key1": b"xyz", "key2": duration("2h"), "key4": true, "key3": 2.71828}`,
```

Comment:
Some types are missing here, same for below list case.

Reply:
Fixed, type coverage should be good now.